AssemblerLib/LocalToGlobalIndexMap.h

```diff
@@ -31,32 +35,31 @@ namespace AssemblerLib
 class LocalToGlobalIndexMap
 {
 public:
+#ifdef USE_PETSC
+    typedef MathLib::RowColumnIndices<PetscInt> RowColumnIndices;
```

Comment:
I see you mean size_t. I can try a test to pass size_t type integer to PetscInt to see whether there is any problem.